File: python/004.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import sys

def p4(n):
    return max(filter(lambda s: str(s) == str(s)[::-1],
                      [ x * y for x in range(1, n) for y in range(x, n) ]))

def main():
    print(p4(999))

if __name__ == '__main__':
    main()

File: elixir/004.exs
#!/usr/bin/env elixir

# https://projecteuler.net/problem=4

Enum.to_list(for x <- 1..999, y <- x..999, do: x * y)
|> Enum.map(&(Integer.to_string(&1)))
|> Enum.filter(&( String.reverse(&1) == &1 ))
|> Enum.map(&(String.to_integer(&1)))
|> Enum.max
|> IO.inspect
